Tests/RandomAccessCollectionTests.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+import XCTest
+
+class RandomAccessCollectionTests: XCTestCase {
+    func testEnumerateWithNext() {
+        let ary = [1,2,3]
+        for (current, next) in ary.enumerateWithNext() {
+            XCTAssert(current == 1)
+            if let next = next {
+                XCTAssert(next == 2)
+            }
+            break
+        }
+    }
+
+    func testEnumerateWithPrevious() {
+        let ary = [1,2,3]
+        for (previous, current) in ary.enumerateWithPrevious() {
+            if current == 1 { continue }
+            XCTAssert(current == 2)
+            if let previous = previous {
+                XCTAssert(previous == 1)
+            }
+            break
+        }
+    }
+}
```

Comment:
I guess it would be smart to add a test on the cases where it could crash, so on `testEnumerateWithNext` on the last item. The previous one you have covered cause you test for the first iteration :v: